src/stock/containers/edit-stock-page/edit-stock-page.container.ts

```diff
@@ -1,39 +1,44 @@
 import {Component, OnInit, OnDestroy} from "@angular/core";
 import {Wine} from "../../entities/Wine";
 import {ActivatedRoute, Router} from "@angular/router";
-import {Subscription, Observable} from "rxjs";
+import {Subscription, Observable, ConnectableObservable} from "rxjs";
 import {StockSandbox} from "../../stock.sandbox";
 @Component({
     selector: "edit-stock-page",
-    template: `
-    <default-page>
-        <main>
-            <div class="row">
-                <div class="col-sm-12">
-                    <h1><i class="fa fa-pencil"></i>&nbsp;Edit wine</h1>
-                </div>
-             </div>
-             <div class="row" *ngIf="(editWine$|async)">
-                <detail-wine-form [wine]="editWine$|async" (onSave)="onSave($event)"></detail-wine-form>
+    template: `<default-page>
+    <main>
+        <div class="row">
+            <div class="col-sm-12">
+                <h1><i class="fa fa-pencil"></i>&nbsp;Edit wine</h1>
             </div>
-        </main>
-    </default-page>
+        </div>
+        <div class="row" *ngIf="(editWine$|async)">
+            <detail-wine-form [wine]="editWine$|async" (onSave)="onSave($event)"></detail-wine-form>
+        </div>
+    </main>
+</default-page>
      `
 })
-export class EditStockPageContainer {
+export class EditStockPageContainer implements OnInit{
     id = this.route.snapshot.params["id"];
-    editWine$ = this.sb.isAuthenticated$
+    editWine$: ConnectableObservable<Wine> = this.sb.isAuthenticated$
```

Comment:
Than that would be an extra piece of content to the course, now I think of it. lets leave the slides as is